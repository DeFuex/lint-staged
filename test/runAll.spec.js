/* eslint no-console: 0 */

import stripColors from 'strip-ansi'
import runAll from '../src/runAll'

describe('runAll', () => {
    it('should work', async () => {
        const config = {
            verbose: true,
            linters: {
                '*': 'eslint'
            }
        }
        let stdout = []
        console.log = jest.fn()
            .mockImplementation(input => stdout.push(JSON.stringify(stripColors(input))))
        await runAll({}, config)
        stdout = stdout.map(line => line && line.replace(/\[\d\d:\d\d:\d\d\]\W/, ''))
        expect(stdout).toMatchSnapshot()
    })
})
