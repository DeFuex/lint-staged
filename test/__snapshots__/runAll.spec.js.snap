// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runAll should work 1`] = `
Array [
  "{\\"verbose\\":true,\\"linters\\":{\\"*\\":\\"eslint\\"},\\"gitDir\\":\\".\\",\\"renderer\\":\\"verbose\\",\\"concurrent\\":true}",
  "\\"Running tasks for * [started]\\"",
  "\\"Running tasks for * [skipped]\\"",
  "\\"â†’ No staged files match *\\"",
]
`;
